src/main/java/de/chojo/repbot/listener/StateListener.java

```diff
@@ -94,6 +94,9 @@ public void onChannelDelete(@NotNull ChannelDeleteEvent event) {
     public void onEmojiRemoved(@NotNull EmojiRemovedEvent event) {
         var guildSettings = guilds.guild(event.getGuild()).settings();
         if (!guildSettings.thanking().reactions().reactionIsEmote()) return;
+
+        guildSettings.thanking().reactions().remove(event.getEmoji().getId());
+
```

Comment:
This check should be before the first check. The first check is false when the main reaction is a unicode emote. Therefore this would only work when the main reaction is a custom emote